libs/qa/comparitor.py

```diff
@@ -0,0 +1,216 @@
+
+
+class Comparitor(object):
+    def __init__(self, snapshot1, snapshot2, api1, api2, state, intervention, fips=None):
+        self.snapshot1 = snapshot1
+        self.snapshot2 = snapshot2
+        self.state = state
+        self.fips = fips
+        self.county = None
+        self.intervention = intervention
+        # TODO (sgoldblatt) swap out here if it's counties
+        self.api1 = api1
+        self.api2 = api2
+        self._date_to_data = {}
+        self._results = []
+        self._generate_date_dictionary()
+
+    def _generate_helper(self, full_data, path, snapshot):
+        for data in full_data[path]:
+            date_of_data = data["date"]
+
+            current_data = self._date_to_data.get(date_of_data, {})
+            current_data[snapshot] = current_data.get(snapshot, {})
+
+            current_data[snapshot][path] = data
+            self._date_to_data[date_of_data] = current_data
+
+    def _generate_last_updated_helper(self, full_data, snapshot):
+        current_data = self._date_to_data.get("current", {})
+        current_data[snapshot] = current_data.get(snapshot, {})
+
+        current_data[snapshot]["projections"] = full_data["projections"]
+        current_data[snapshot]["actuals"] = full_data["actuals"]
+        self._date_to_data["current"] = current_data
+
+    def _generate_date_dictionary(self):
+        """
+            {"2020-01-27": {
+                271: {
+                    "actualTimeseries": {'population': None, 'intervention': 'STRONG_INTERVENTION', 'cumulativeConfirmedCases': 2, 'cumulativePositiveTests': None, 'cumulativeNegativeTests': None, 'cumulativeDeaths': None, 'hospitalBeds': {'capacity': None, 'totalCapacity': None, 'currentUsageCovid': None, 'currentUsageTotal': None, 'typicalUsageRate': None
+                    }, 'ICUBeds': {'capacity': None, 'totalCapacity': None, 'currentUsageCovid': None, 'currentUsageTotal': None, 'typicalUsageRate': None
+                    }, 'date': '2020-01-26'
+                    }, 
+                    ...
+                    "timeseries": {...}
+                }, 
+                286: {},
+            }}
+        """
+        for api, snapshot in [(self.api1, self.snapshot1), (self.api2, self.snapshot2)]:
+            self._generate_helper(api, "actualsTimeseries", snapshot)
+            self._generate_helper(api, "timeseries", snapshot)
+
+        for api, snapshot in [(self.api1, self.snapshot1), (self.api2, self.snapshot2)]:
+            self._generate_last_updated_helper(api, snapshot)
+
+    def _getMetricValue(self, metric: QAMetric, date: str, snapshot: int, isProjected: bool):
+        metric_path = metric.projection_path if isProjected else metric.actual_path
+        if not metric_path:
+            return None
+        data_at_date = self._date_to_data.get(date)
+        if not data_at_date:
+            capture_event(
```

Comment:
Can you add a logging warning here as well?

Reply:
yeah also this didn't seem to do the thing so messing around a bit with it now

Reply:
okay fixed